apps/prairielearn/src/pages/authLogin/authLogin.html.ts

```diff
@@ -114,17 +114,18 @@ export function AuthLogin({ institutionAuthnProviders, service, resLocals }: Aut
           <div class="login-container">
             <div>
               <h1 class="text-center">PrairieLearn</h1>
-              <h2 class="text-center subheader">
+              <h2 class="text-center subheader mb-5">
                 Sign in ${service ? `to continue to ${service}` : ''}
               </h2>
-              <div class="login-methods mt-5">
-                ${resLocals.devMode
-                  ? html`
-                      <a class="btn btn-success w-100" href="/pl/dev_login" role="button">
-                        <span class="font-weight-bold">DevMode by-pass</span>
-                      </a>
-                    `
-                  : null}
+              ${resLocals.devMode
+                ? html`
+                    ${DevModeBypass()}
+                    <hr />
+                    ${DevModeLogin({ csrfToken: resLocals.__csrf_token })}
+                    <hr />
+                  `
+                : null}
```

Comment:
Can we use `''` here rather than `null`? I know this isn't a change introduced in this PR, but I'd prefer to keep our templating free of magic conversions whenever possible. Just as we removed the magic handling of booleans, I'd like to strip out the magic handling of `null` in `html`-templated strings as well.

Reply:
Interesting - and then render them directly as the string `null` in the output? Are there actual uses for that? Treating `null` as "render nothing" is very common across templating engines, including [Lit](https://lit.dev/docs/templates/conditionals/#conditionally-rendering-nothing) and [React](https://react.dev/learn/conditional-rendering#conditionally-returning-nothing-with-null).

Reply:
> Treating `null` as "render nothing" is very common across templating engines, including [Lit](https://lit.dev/docs/templates/conditionals/#conditionally-rendering-nothing) and [React](https://react.dev/learn/conditional-rendering#conditionally-returning-nothing-with-null).

And mustache and EJS...

Reply:
My current opinion is that it should be a type error to interpolate `null` into an `html`-templated string. I haven't come across any place where it's actually a good idea to allow `null` but treat it as `''`. Every case I've seen is like the code above, where we could just as easily use the explicit empty string as the value rather than `null`. For all the cases where we have a nullable variable that we want to render, the case where it is `null` should almost always be handled specially in any case and having the type system catch this would have prevented bugs.

For example, consider a variable like `webhook_url` which can be NULL (if we don't have a webhook) and a string if we do. I would like the template `html`<a href="${webhook_url}">${webhook_url}</a>` to give a type error because the `null` case is not allowed. It's currently easy to write code like this and forget that the `null` case should be handled specially.

Reply:
Changed in ce534b6.

Reply:
Thanks!